spec/01-unit/11-reports_spec.lua

```diff
@@ -2,11 +2,32 @@ local meta = require "kong.meta"
 local helpers = require "spec.helpers"
 local cjson = require "cjson"
 
-
 describe("reports", function()
   local reports, bytes, err
+  local expected_data = "version"
   local port = 8189
-  local opts = { tls = true }
+
+  setup(function()
+    -- start the echo server
+    assert(helpers.start_kong({
+               nginx_conf = "spec/fixtures/custom_nginx.template",
+               -- we don't actually use any stream proxy features in tcp_server,
+               -- but this is needed in order to load the echo server defined at
+               -- nginx_kong_test_tcp_echo_server_custom_inject_stream.lua
+               stream_listen = helpers.get_proxy_ip(false) .. ":19000",
+               -- to fix "Database needs bootstrapping or is older than Kong 1.0" in CI.
+               database = "off",
+               log_level = "info",
+                      }))
+
+    assert(helpers.is_echo_server_ready())
```

Comment:
I have not seen us using `is_kong_<any>_server_ready` in a past. Is there reason for this? Or how it works elsewhere in tests where we just start kong? Is it covered by timeout of admin/proxy client in helpers? But shouldn't it then be covered with reports client too?

Reply:
In the original thread based `tcp_server()`, there is a block of code that does the same function to check whether the echo server ready.